mcp/sensors/star_camera_trigger.c

```diff
@@ -0,0 +1,231 @@
+        sc2_trigger.trigger = 1;
+        return 1;
+    } else if (AZ_VEL_LIMIT > PointingData[1].gy_az && PointingData[1].gy_az > -AZ_VEL_LIMIT
+     && CommandData.sc_trigger.enable_sc_gyro_trigger == 1) {
+        sc1_trigger.trigger = 1;
+        sc2_trigger.trigger = 1;
+        return 1;
+    } else if (AZ_VEL_LIMIT > PointingData[2].gy_az && PointingData[2].gy_az > -AZ_VEL_LIMIT
+     && CommandData.sc_trigger.enable_sc_gyro_trigger == 1) {
+        sc1_trigger.trigger = 1;
+        sc2_trigger.trigger = 1;
+        return 1;
+    } else {
+        sc1_trigger.trigger = 0;
+        sc2_trigger.trigger = 0;
+        return 0;
+    }
+}
+
+
+/**
+ * @brief checks the command data structure for an updated timeout value.
+ * 
+ * @return int number of seconds to set the timeout to, 2 if
+ * we do not update, whatever we tell this if we do update.
+ */
+static int sc_timeout_update(void) {
+    if (CommandData.sc_trigger.starcam_image_timeout_update == 1) {
+        CommandData.sc_trigger.starcam_image_timeout_update = 0;
+        return CommandData.sc_trigger.starcam_image_timeout;
+    } else {
+        return SC_STANDARD_TIMEOUT_SEC;
+    }
+}
+
+
+/**
+ * @brief Sets up the thread which monitors the velocity to send star camera triggers
+ * to the star camera computers.
+ * 
+ * @param args void* cast socket data that contains IP addr and port number to connect to
+ */
+void *star_camera_trigger_thread(void *args) {
+    struct socket_data * socket_target = args;
+    static int first_time = 1;
+    int sleep_interval_usec = SC_GYRO_CHECK_INTERVAL_USEC; // 10ms intervals to check the gyro data
+    int sleep_photo_interval_sec = SC_STANDARD_TIMEOUT_SEC;
+    // after the turnaround trigger we wont try to trigger for 2 seconds
+    int sockfd;
+    struct addrinfo hints;
+    struct addrinfo *servinfo;
+    struct addrinfo *servinfoCheck;
+    int returnval;
+    int bytes_sent;
+    int length;
+    int *retval;
+    char message_str[40];
+    char ipAddr[INET_ADDRSTRLEN];
+    int sending = 1;
+    int which_sc;
+    int packet_status = 0;
+    struct star_cam_trigger *scTrigger;
+    // set the status of this socket to 1 (started)
+    if (!strcmp(socket_target->ipAddr, SC1_IP_ADDR)) {
+        scTrigger = &sc1_trigger;
+        which_sc = 1;
+        snprintf(message_str, sizeof(message_str), "%s", "SC1 trigger thread");
+        blast_info("Trigger socket pointed at SC1\n");
+    } else if (!strcmp(socket_target->ipAddr, SC2_IP_ADDR)) {
+        scTrigger = &sc2_trigger;
+        which_sc = 2;
+        snprintf(message_str, sizeof(message_str), "%s", "SC2 trigger thread");
+        blast_info("Trigger socket pointed at SC2\n");
+    } else {
+        blast_info("IP provided is %s, expected is %s or %s\n", socket_target->ipAddr, SC1_IP_ADDR, SC2_IP_ADDR);
+        blast_err("Invalid ip target for star camera\n");
+        return NULL;
+    }
+    while (sending) {
+        sending = check_sc_thread_bool(which_sc);
```

Comment:
It looks like a `sc1_interrupt_param` or `sc2_interrupt_param` command will terminate this thread until `mcp` is restarted. Is this what you intended?

Reply:
Yes, the use cases are few but I wanted an option to kill the threads (and left this in specifically to kill another "param")